### PR TITLE
Library guidelines restructuring

### PR DESCRIPTION
**Pull request by daniCsorbaJB**

### CODE REVIEW

File: docs/api-guidelines.tree [hunk 1]
```diff
@@ -4,13 +4,12 @@
 
 <product-profile id="kl"
                  name="API guidelines"
-                 start-page="jvm-api-guidelines-introduction.md">
+                 start-page="api-guidelines-introduction.md">
 
     <chunk include-id="api-guidelines">
-        <toc-element id="jvm-api-guidelines-introduction.md"/>
-        <toc-element id="jvm-api-guidelines-readability.md"/>
```
Comment (koshachy) [inline docs/api-guidelines.tree:11]: What about reditrecting other pages?
For now these links will be just dead

File: docs/topics/api-guidelines-backward-compatibility.md [hunk 2]
```diff
@@ -0,0 +1,318 @@
...
+The `apiCheck` task is invoked at build time by the standard Gradle `check` task.
+When compatibility is broken, the build fails. At that point, you should run the `apiDump` task manually and compare
+the differences between the old and new versions.
+If you are satisfied with the changes, you can update the existing `.api` file, which resides within your VCS.
+
+The validator has [experimental support for validating KLibs](https://github.com/Kotlin/binary-compatibility-validator?tab=readme-ov-file#experimental-klib-abi-validation-support) produced by multiplatform libraries.
+
+## Specify return types explicitly
+
+As discussed in the [Kotlin coding guidelines](https://kotlinlang.org/docs/coding-conventions.html#coding-conventions-for-libraries),
```
Comment (koshachy) [inline docs/topics/api-guidelines-backward-compatibility.md:56]: I think that relative links should work here (coding-conventions.md)

Comment (koshachy) [reply docs/topics/api-guidelines-backward-compatibility.md:56]: Look at other integrated docs, I guess you'll find the answer

File: docs/topics/api-guidelines-backward-compatibility.md [hunk 3]
```diff
@@ -0,0 +1,318 @@
...
+       …
+```
+
+This is because the signature of the method changed in the bytecode generated by the Kotlin/JVM compiler, breaking binary compatibility.
+
+Source compatibility, however, is preserved. If you recompile both files, the program will run as before.
+
+### Use overloads to preserve compatibility {initial-collapse-state="collapsed"}
+
+When writing Kotlin code for the JVM, you can use the [`@JvmOverloads`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin.jvm/-jvm-overloads/) annotation on functions with default arguments.
```
Comment (koshachy) [inline docs/topics/api-guidelines-backward-compatibility.md:139]: But for API reference links should be absolute (I mean, don't touch them :D )

File: docs/topics/api-guidelines-backward-compatibility.md [hunk 4]
```diff
@@ -0,0 +1,318 @@
...
+that accepts default arguments to preserve binary compatibility. In the example above, this means creating a separate 
+`fib` function for the case where you wish to take an `Int` parameter:
+
+```kotlin
+fun fib() = … 
+fun fib(input: Int) = …
+```
+
+## Avoid widening or narrowing return types
+
```
Comment (koshachy) [inline docs/topics/api-guidelines-backward-compatibility.md:168]: Nit-pick: double empty line

File: docs/topics/api-guidelines-backward-compatibility.md [hunk 5]
```diff
@@ -0,0 +1,318 @@
...
+If you need to call an internal API of your library from an inlined public function, you can do so by annotating it with [`@PublishedApi`](https://kotlinlang.org/api/latest/jvm/stdlib/kotlin/-published-api/).
+This makes the internal declaration effectively public, as references to it will end up in compiled client code.
+Therefore, it must be treated the same as public declarations when making changes to it, as these changes might affect binary compatibility.
+
+## Evolve APIs pragmatically
+
+There are cases where you need to make breaking changes to your library’s API over time by removing or changing an existing declaration.
+In this section, we’ll discuss how to handle such cases pragmatically.
+
+When users upgrade to a newer version of your library, they should not end up with unresolved references to your library’s
```
Comment (koshachy) [inline docs/topics/api-guidelines-backward-compatibility.md:282]: find+replace ’ to '

File: docs/topics/api-guidelines-informative-documentation.md [hunk 6]
```diff
@@ -0,0 +1,158 @@
...
+Every supported API entry point should be documented using [KDoc](https://kotlinlang.org/docs/kotlin-doc.html).
+
+Kotlin’s documentation engine, [Dokka](https://kotlinlang.org/docs/dokka-introduction.html), includes only public declarations in its outputs by default. As discussed in the [Simplicity](api-guidelines-minimizing-mental-complexity.md#simplicity) section,
+you should minimize your public API and remove public entry points that you don’t want users to access.
+If there are APIs you cannot hide from users by controlling their visibility, omit them from the documentation using the [suppress directive](https://kotlinlang.org/docs/kotlin-doc.html#suppress).
+
+Begin the description of entry points with a clear, high-level description of what the function does.
+Avoid simply restating the signature in natural language.
+
+For example, don’t say “takes a `String` and returns a `Connection`” but instead perhaps
```
Comment (koshachy) [inline docs/topics/api-guidelines-informative-documentation.md:86]: find+replace ” to "

File: docs/topics/api-guidelines-minimizing-mental-complexity.md [hunk 7]
```diff
@@ -0,0 +1,685 @@
...
+and project-specific conventions.
+
+A Kotlin library should also provide a suite of unit and integration tests covering all documented behaviors of all the API entry points.
+Tests should include a wide range of inputs, especially known boundary and edge cases. Any untested behavior should be assumed to be (at best) unreliable.
+
+Use this suite of tests during development to verify that changes do not break existing behavior.
+Run these tests on every release as part of a standardized build and release pipeline.
+Tools like [Kover](https://github.com/Kotlin/kotlinx-kover) can be integrated into your build process to measure coverage and generate reports.
+
+## Predictability
```
Comment (koshachy) [inline docs/topics/api-guidelines-minimizing-mental-complexity.md:343]: Why did you decide to leave it on one page?
Out of curiosity

Comment (koshachy) [reply docs/topics/api-guidelines-minimizing-mental-complexity.md:343]: I mean Predictabiliy, Debuggability...

**Review by koshachy — CHANGES_REQUESTED** [review]
* Swap to proper symbols " and `
* Relative links to kotlinlang/docs (not kotlinlang/api)
* Split to more pages (like it was before) - to discuss
* Redirects for deleted pages

Some minor markup things could be fixed after the KotlinConf.

File: docs/topics/api-guidelines-backward-compatibility.md (hunk 5 above)
Comment (daniCsorbaJB) [reply docs/topics/api-guidelines-backward-compatibility.md:282]: ah nice catch! - replaced all occurences

File: docs/topics/api-guidelines-backward-compatibility.md (hunk 2 above)
Comment (daniCsorbaJB) [reply docs/topics/api-guidelines-backward-compatibility.md:56]: good point - I checked it seems to be working - I replaced all occurences

File: docs/topics/api-guidelines-minimizing-mental-complexity.md (hunk 7 above)
Comment (daniCsorbaJB) [reply docs/topics/api-guidelines-minimizing-mental-complexity.md:343]: as discussed - it's still open - let's revisit later! :)

**Review by koshachy — APPROVED** [review]
LGTM. I hope it will not trigger the master branch deploy :) 
Double-check it please :D
